src/ast/CommentDecl.ts

```diff
@@ -11,8 +11,19 @@ export default class CommentDecl extends AstNode {
     comments: string[];
 
     public parse(context: Tokenizer): any {
-        // TODO: implement the rest.
+        // We start by checking "[" here, since when we call this, we would have consumed the "comment" token already
```

Comment:
I don't think we should consume the comments token before this node, we should do a checkToken in the higher-level node, but CommentDecl should be the one to consume it.

Reply:
Okay, that makes sense to me. Thanks 🌮

Reply:
NVM!